src/blocks/battery.rs

```diff
@@ -198,12 +207,25 @@ impl BatteryDevice for PowerSupplyDevice {
     }
 
     fn power_consumption(&self) -> Result<u64> {
+        // power consumption in µWh
         let power_path = self.device_path.join("power_now");
+        // current consumption in µA
+        let current_path = self.device_path.join("current_now");
+        // voltage in µV
+        let voltage_path = self.device_path.join("voltage_now");
 
         if power_path.exists() {
             Ok(read_file("battery", &power_path)?
                 .parse::<u64>()
                 .block_error("battery", "failed to parse power_now")?)
+        } else if current_path.exists() && voltage_path.exists() {
+            let current = read_file("battery", &current_path)?
+                .parse::<u64>()
+                .block_error("battery", "failed to parse current_now")?;
+            let voltage = read_file("battery", &voltage_path)?
+                .parse::<u64>()
+                .block_error("battery", "failed to parse voltage_now")?;
+            Ok(current * voltage)
```

Comment:
```suggestion
            Ok((current * voltage) / 1000000)
```
Because the result is in microwatts otherwise.

(This doesn't actually fully make sense, since microamps * microvolts should be picowatts, not microwatts. But the actual value suggests it's actually microwatts, so maybe current is in milliamps and voltage in millivolts?)